src/main/kotlin/net/maizegenetics/phgv2/cli/AgcCompress.kt

```diff
@@ -0,0 +1,228 @@
+            throw IllegalArgumentException("Error running ProcessBuilder for agc create or append: ${exc.message}")
+        }
+
+        return true
+    }
+
+    // This function takes a list of fasta file names and returns a list of sample names
+    // Sample names are the file names without extension
+    fun getCurrentSampleNames(fastaFiles:List<String>):List<String> {
+        //create a list of names from the fasta files, where the name is
+        // just the file name, no path, and without the extension
+        val sampleNames = mutableListOf<String>()
+        fastaFiles.forEach { sampleNames.add(it.split("/").last().split(".").first()) }
+        return sampleNames
+    }
+
+    fun getSampleListFromAGC(agcFile:String,tempDir:String): List<String> {
+        // This function will return a list of samples from the AGC compressed file.
+        // This will be used to verify that the new list of fastas has nothing overlapping
+        // the exsiting fastas in the AGC compressed file.
+
+        var sampleList = listOf<String>()
+        // Query the agc compressed file to get list of sample names
+        var builder = ProcessBuilder("conda","run","-n","phgv2-conda","agc","listset",agcFile)
+        var redirectOutput = tempDir + "/agc_create_output.log"
+        var redirectError = tempDir + "/agc_create_error.log"
+        builder.redirectOutput( File(redirectOutput))
+        builder.redirectError( File(redirectError))
+
+        myLogger.info("begin Command to list agc sample names:" + builder.command().stream().collect(Collectors.joining(" ")))
+        try {
+            var process = builder.start()
+            var error = process.waitFor()
+            if (error != 0) {
+                println("agc listset run via ProcessBuilder returned error code $error")
+                throw IllegalArgumentException("Error running ProcessBuilder to list agc samples: ${error}")
+            }
+            // read the output file to get the list of samples
+            sampleList = File(redirectOutput).readLines()
+
+        } catch (exc: Exception) {
+            myLogger.error("Error: could not list agc samples.")
+            throw IllegalArgumentException("Error running ProcessBuilder for agc listset: ${exc.message}")
+        }
+
+        return sampleList
+    }
+
+    // This function will take a list of fasta names compiled from the input fasta
+    // files and compare them to those name already in the agc compressed file.
+    // It returns a list of any duplicates
+    fun compareNewExistingSampleNames(agcFile:String, newFastas:List<String>, tempDir:String): List<String> {
+        // Need to process the newFastas list to just the name without extension or path
+        // that is what is stored as the sample name in the AGC compressed file.
+        val newSampleNames = mutableListOf<String>()
+        newFastas.forEach { newSampleNames.add(it.split("/").last().split(".").first()) }
```

Comment:
call getCurrentSampleNames on this to reduce code copying.

Reply:
done